pkg/logql/log/filter.go

```diff
@@ -14,6 +14,32 @@ import (
 	"github.com/grafana/loki/pkg/util"
 )
 
+// LineMatchType is an enum for line matching types.
+type LineMatchType int
+
+// Possible LineMatchTypes.
+const (
+	LineMatchEqual LineMatchType = iota
+	LineMatchNotEqual
+	LineMatchRegexp
+	LineMatchNotRegexp
+)
```

Comment:
We probably could omit the `Line` prefix, but that would likely lead to confusion with the label match types.

Reply:
yes I agree